### ISSUE TITLE
Issues related to 'Display name' resolving

### ISSUE THREAD

**Issue by VolodLytvynenko** [status-im/status-react#15500]
### Problem:
If a new user is just created, then his 'display name' is not resolved immediately if to open his profile. Sometimes it takes 3-5 minutes

### Preconditions:
User_B:  sets display name during onboarding

### Steps to reproduce:
1. User_A: Go to 'add contact' page -> Paste chat key of User_B into 'ENS or Chat key' 
2. Tap "view profile" button
3. Check contact name resolving

### Actual result:
- "Display name" of newly created contact is not resolved on the 'User found' section and on the profile page
![image](https://user-images.githubusercontent.com/52490791/227981064-c3d7992c-4adc-495b-8328-47b470e31619.png)

- "Display name" of newly created contact is not resolved on the opened user profile
![image](https://user-images.githubusercontent.com/52490791/227981268-260fe2da-e833-442e-9cd6-136440909907.png)

https://user-images.githubusercontent.com/52490791/227979653-60b9c6a6-80e2-4a53-888d-8596025324f5.mp4

### Expected result:
The display name of the contact is shown

### Additional info:
This issue is reproducible in cases if User_A pastes the chat key of User_B which has just been created (1-3 minutes ago)

**Comment by VolodLytvynenko** [comment]
## <ins>[FIXED]</ins> ISSUE 2: 'Display name' is not resolved for message sender in chat  after the message is sent
#### Steps to reproduce:
1. Open any chat
2. Send the message
3. Check your name

#### Actual result:
the 3 random name is shown
![image](https://user-images.githubusercontent.com/52490791/227989138-acb38354-3b74-4c72-be92-cd4c311d05d8.png)

#### Expected result:
The user sees his display name after the message is sent

**Comment by VolodLytvynenko** [comment]
## ISSUE 3: 'Display name' is not resolved after the link of the profile is shared
#### Steps to reproduce:
1. Go to the profile page
2. Tap on the 'chat key' -> tap 'Share Link'  
3. Send the link somewhere (for example: in the telegram)
4. Open the link 

#### Actual result:
The '3 random name' is shown in the link preview for other services (example: in telegram)
![image](https://user-images.githubusercontent.com/52490791/228004533-532ca62b-64b1-4af2-986c-5a85177ab8ba.png)

The '3 random name' is shown in the opened link
![image](https://user-images.githubusercontent.com/52490791/228004994-231bce9b-affc-41d9-ac70-2d30bdad509a.png)


#### Expected result:
The 'display name' is resolved in the link preview and if the link is opened

**Comment by VolodLytvynenko** [comment]
## <ins>[FIXED]</ins> ISSUE 4: "Display name" is not resolved in the contact request acceptance toast
#### Steps to reproduce:
1. Create User_A and User_B with Display names during onboarding
2. User_A sends requests to User_B
3. User_B accepts contact request in AC
4. Check toast as User_A

#### Actual result:
The '3 random name' is shown instead of 'Display name' in the toast notifying accepted contact request

![image](https://user-images.githubusercontent.com/52490791/228294660-ab649296-6fdc-4e9a-9209-8bdd4c824713.png)

#### Expected result:
'Display name' is shown in the accepted contact request toast

**Comment by churik** [comment]
## <ins>[FIXED]</ins> ISSUE 5: "Display name" is not resolved in mentions in AC
#### Steps to reproduce:
 - User A and User B join the same community
 - User A: mention User B in community
 - User A: check AC
 
 #### Actual result:
![image](https://user-images.githubusercontent.com/4557972/228813673-3288cf15-fc01-4605-b471-d98ca776602b.png)
 
 
#### Expected result:
'Display name' is shown

**Comment by churik** [comment]
Note: 
Issues 1,2,5 should be resolved in 1.22, they are blockers for e2e.

**Comment by pavloburykh** [comment]
Reopened the ticket as [ISSUE 1](https://github.com/status-im/status-mobile/issues/15500#issue-1642331257) still not resolved and it is e2e blocker. 

[ISSUE 3](https://github.com/status-im/status-mobile/issues/15500#issuecomment-1485455153) is not resolved as well

**Comment by churik** [comment]
Also, there is another issue - even if the display name is set when searching by ENS name, the still old 3-random name is shown
![IMAGE 2023-04-14 15:29:36](https://user-images.githubusercontent.com/4557972/232057103-d2a9c989-cbea-4fc2-8ee4-cae7a86b2d48.jpg)

If it is possible, would be nice to avoid displaying a 3-random name, which is not used anymore. 
ENS name is picked up only after receiving a message from this user

**Comment by yqrashawn** [comment]
working on ISSUE 1
can reproduce it

seems the display name got updated right after received the new contact message from status-go 

need to check if we can speed up this in status-go

**Comment by cammellos** [comment]
@yqrashawn this is due to the fact that we resolve ens names asynchronously, to avoid getting spammed or having to wait for ENS response before showing the message to the user.
It could also be solved via UX, where we show the ENS name, but we show it as "unverified" (opacity etc), feel free to contact the design team if you think that's a viable solution.

**Comment by yqrashawn** [comment]
Updates on ISSUE1

So, there's nothing wrong on the receiver(`User_A`) side, the request to get contact info is fast, the problem is more on the publishing side. Thanks to @cammellos for help me to debug status-go
I plan to put issue1 aside and work on issue3 first, since it has higher priority.

**Comment by yqrashawn** [comment]
Updates on ISSUE3

code at https://github.com/status-im/universal-links-handler/blob/master/routes/index.js
might need to use js-waku to get user display-name https://github.com/waku-org/js-waku

**Comment by yqrashawn** [comment]
updates 

there's already implmentation at https://github.com/status-im/status-web/tree/59276f941fb4b9f96a229559286113d24fd8bcb0/packages/status-js/src/request-client/request-client.ts
but @status-im/js is not targeting Node.js

plan is to use code bundler and transformer to enable using it in Node.js

**Comment by yqrashawn** [comment]
updates

able to run @status-im/js in node.js with webpack
`Error: Timed out waiting for a remote peer.`
 
code at branch https://github.com/status-im/universal-links-handler/tree/feat/display-name-from-waku

**Comment by yqrashawn** [comment]
back to this issue

seems [ISSUE4](https://github.com/status-im/status-mobile/issues/15500#issuecomment-1487180548) is fixed now as long as User_B is not a newly created user @VladimrLitvinenko 
newly created user display name is still not synced fast enough, which is ISSUE 1

**Comment by yqrashawn** [comment]
same on ISSUE 5

**Comment by VolodLytvynenko** [comment]
Hi, @yqrashawn correct. Issues 4,5 are fixed. Thank you

**Comment by yqrashawn** [comment]
Forget to update this here.
So for situations where contact display name is not synced yet, we use public key as a fallback

![image](https://github.com/status-im/status-mobile/assets/15090582/bfc84a04-fe59-44f8-927b-13fb1da02013)


> The fallback to the username is the chatkey (the short version) as we did here:
> https://www.figma.com/file/7KIYbhoqNGAIFonE0w9TDz/Messages-for-Mobile?type=design&node-id=8291%3A514145&t=PxdzxancUk8DEj3S-1

**Comment by yqrashawn** [comment]
also affects user-avatar in contact view https://github.com/status-im/status-mobile/pull/16193#issuecomment-1594485265

better check all user-avatars

**Comment by churik** [comment]
More issues on the matter: 
now all contacts from waku backup and after sync are restored with 3 random-names

**Comment by yqrashawn** [comment]
unassign myself
I'm still working on the avatar PR, will go back here and pick up this after that PR and if this is still available

**Comment by yqrashawn** [comment]
contact 3-random-word name is coming from status-go when `initialize-contacts`
we need to somehow mark them in status-go and handle it in status-mobile
or we can set primary name to shortened compressed pub key, since dot is not allowed when set display name

**Comment by qoqobolo** [comment]
### ISSUE 6: Display names are not resolved in mentions in push notifications (Android)

**Steps:**
1. UserA: enable PNs and put the app to background
2. UserB: mention UserA in the community
3. UserA: check the PN (3-random name is displayed instead of the display name)

Sender: 
[test_community_mentions_push_notification_geth2.log](https://github.com/status-im/status-mobile/files/12280685/test_community_mentions_push_notification_geth2.log)
Receiver: 
[test_community_mentions_push_notification_geth1.log](https://github.com/status-im/status-mobile/files/12280686/test_community_mentions_push_notification_geth1.log)


<img width="502" alt="Screenshot 2023-08-07 at 15 51 22" src="https://github.com/status-im/status-mobile/assets/67952253/74691194-4e12-4897-8c0e-91362db500b8">

**Comment by VolodLytvynenko** [comment]
Here's a summarized update on the status of the reported issues as of the date of this message
- ISSUE 1:  https://github.com/status-im/status-mobile/issues/15500#issue-1642331257 **Not yet fixed.** 
- ISSUE 2:  https://github.com/status-im/status-mobile/issues/15500#issuecomment-1485346552 **Fixed**
- ISSUE 3: https://github.com/status-im/status-mobile/issues/15500#issuecomment-1485455153 **Not yet  fixed.** 
- ISSUE 4: https://github.com/status-im/status-mobile/issues/15500#issuecomment-1487180548 **Fixed**
- ISSUE 5:  https://github.com/status-im/status-mobile/issues/15500#issuecomment-1490091274 **Fixed**
- ISSUE 6:  https://github.com/status-im/status-mobile/issues/15500#issuecomment-1667923193 **Not yet  fixed.** 
- ISSUE 7: https://github.com/status-im/status-mobile/issues/15500#issuecomment-1651964037 **Fixed**

**Comment by cammellos** [comment]
@VolodLytvynenko Issue 3 is a won't fix, it's a change needed on the website, we are just unfurling the content provided

@churik is issue 7 still relevant?

**Comment by VolodLytvynenko** [comment]
> @VolodLytvynenko Issue 3 is a won't fix, it's a change needed on the website, we are just unfurling the content provided
> 
> @churik is issue 7 still relevant?

hi @cammellos issue 7 is not reproducible for me anymore. Thank you!

**Comment by cammellos** [comment]
@churik can we close this? I believe the only one not fixed is the website, but that's not going to be fixed as we are getting rid of it
